### ISSUE TITLE
Questions: implementation + html parser adapter

### ISSUE THREAD

**Issue by azrafe7** [litehtml/litehtml#49]
Hi,
just found out about this lib, and it seems very interesting. What I really like is the render-agnostic approach!

I have a couple of questions that I hope you can answer:

 1. am I right in thinking that gumbo is only used for html parsing, and this lib leverages it by in turn taking care of css (computing rendering positions/size/clipping/etc.)?
 2. would you think it would be feasible to adapt the library to use something like BeautifulSoup/JSoup for the html parsing instead of gumbo?
 3. would you mind if I (try to) port the lib to another language (obviously crediting you and including your license)?

Thanks and keep up the good work!

PS: oh... and in any case I'de be really glad if you can manage to implement floats for rendering positions (as suggested in another issue). I see it's not a trivial task though!

**Comment by tordex** [comment]
> 1. am I right in thinking that gumbo is only used for html parsing, and this lib leverages it by in turn taking care of css (computing rendering positions/size/clipping/etc.)?

Yes, gumbo is used to parse html only. CSS are parsed by litehtml itself.

> 2. would you think it would be feasible to adapt the library to use something like BeautifulSoup/JSoup for the html parsing instead of gumbo?

Hmm... this is interesting. I guess this is not a big problem. It is simple to build elements tree from other parser. I'll add this into todo-list :+1: 

> 3. would you mind if I (try to) port the lib to another language (obviously crediting you and including your license)?

I'll be glad to see ports to another languages.

**Comment by azrafe7** [comment]
Great! :+1: 
And nice to hear it sounds interesting to you too.

**Comment by azrafe7** [comment]
And please post here if you make any progress in that direction.

**Comment by azrafe7** [comment]
Hi there, and sorry for bumping this up. Just a couple more questions/observations (which really are feature requests :D).

 - I can't stress enough how much I like the render-agnostic property of this lib, so what I'd hope for is to have **two** more interfaces (like `litehtml::document_container`): one for handling html (say `parseHtml(string)`), and one for parsing/applying css (say `computeStyle(style, doc)`)
 - I get it could be cumbersome to do (as this may also need an intermediate representation), but I still think that separating the processes could make things easier to work with (and also make it possible to switch libraries for the specific needs)

I know I'm speaking out of my direct needs/hopes, but do these requests seem sensible to you and addressable in the nearest future?

PS: maybe this needs to be a separate issue?!
PPS: main reason for asking this is that there are only a few libs out there (in the opensource zone), that return layout props from css, although many offer to parse html and/or css (with no layout). So mixing&matching them might be very interesting (if only for benchmarking purposes and a common task).

Thanks.
